pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -360,13 +375,20 @@ func (cm *containerManagerImpl) setupNode() error {
 
 	// Setup top level qos containers only if CgroupsPerQOS flag is specified as true
 	if cm.NodeConfig.CgroupsPerQOS {
-		qosContainersInfo, err := InitQOS(cm.NodeConfig.CgroupDriver, cm.NodeConfig.CgroupRoot, cm.subsystems)
+		qosContainersInfo, err := InitQOS(cm.NodeConfig.CgroupDriver, cm.cgroupRoot, cm.subsystems)
 		if err != nil {
 			return fmt.Errorf("failed to initialise top level QOS containers: %v", err)
 		}
 		cm.qosContainers = qosContainersInfo
 	}
 
+	// Enforce Node Allocatable (if required)
+	if err := enforceNodeAllocatableCgroups(cm.NodeConfig.NodeAllocatableConfig,
+		cm.getNodeAllocatableInternal(false), // ignore eviction thresholds.
```

Comment:
i thought kubepods needs to have allocatable = capacity - reserved + thresholds enforced.  right now, this appears to not give the buffer discussed.